File: app/models/api/feed.rb
module Api
  class Feed
    def initialize(feed:)
      @feed = feed
      @last_entry = feed.last_entry
      @tags = feed.tags
    end

    attr_reader :feed, :last_entry, :tags

    def attributes
      {
        id: feed.id,
        endpoint: feed.endpoint,
        lastPublishedAt: feed.last_published_at.iso8601,
        title: feed.title,
        lastEntry: last_entry_attirbutes,
        tags: tags_attributes
      }
    end

    private

    def last_entry_attirbutes
      last_entry ? Entry.new(entry: last_entry).attributes : nil
    end

    def tags_attributes
      tags.map { |tag| tag_attributes(tag) }
    end

    def tag_attributes(tag)
      {
        id: tag.id,
        body: tag.body,
        feedId: feed.id
      }
    end
  end
end

File: spec/javascript/mocks/Feed.ts
import { Feed } from "@js/types/types";
import { buildEntry } from "./Entry";
import { buildFeedTag } from "./FeedTag";

export const buildFeed = (id: number): Feed => {
  return {
    id: id,
    title: `feed title ${id}`,
    lastPublishedAt: new Date("2021-05-03T14:35:46+09:00"),
    endpoint: "https://example.com/rss",
    lastEntry: buildEntry(id),
    tags: [buildFeedTag(id), buildFeedTag(id)],
  };
};
